### PR TITLE
plat-stm32mp1: scmi_server: fix case when regulator state is unchanged

### PR DESCRIPTION
**Pull request by etienne-lms**
Fixes the return value when an SCMI regulator is already in the expected state. Prior this change the implementation returned SCMI_GENERIC_ERROR instead of SCMI_SUCCESS.

Changes also the SCMI return code in case of failure from SCMI_GENERIC_ERROR to SCMI_INVALID_PARAMETERS when the requested state is not one of the 2 supported SCMi voltage domain states supported (SCMI_VOLTAGE_DOMAIN_CONFIG_ARCH_ON or
SCMI_VOLTAGE_DOMAIN_CONFIG_ARCH_OFF).

Fixes: 23e200628dad ("plat-stm32mp1: scmi_server: use registered regulators")

### CODE REVIEW

File: core/arch/arm/dts/stm32mp157c-ed1.dts [hunk 1]
```diff
@@ -253,6 +253,7 @@
 			vdd_usb: ldo4 {
 				regulator-name = "vdd_usb";
 				interrupts = <IT_CURLIM_LDO4 0>;
+				regulator-always-on;
```
Comment (jforissier) [inline core/arch/arm/dts/stm32mp157c-ed1.dts:256]: The DT changes should probably go in a separate commit.
I did not quite understand the explanation about this regulator in the commit description :-/ perhaps you could reword it?

**Review by jforissier — COMMENTED** [review]
- For "plat-stm32mp1: scmi_server: fix regulator framework integration": see comments below
- For "plat-stm32mp1: scmi_server: report invalid regulator state request":
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/dts/stm32mp157c-ed1.dts (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/dts/stm32mp157c-ed1.dts:256]: Ok, i'll try :) sorry for my bad phrasing.

**Comment by etienne-lms** [comment]
Thanks for the feedback. I have split the commit in 2 and rephrase the commit message. It's far more verbose now. I hope it clear. Feel free to complain if not.

I have not applied your review tag. I'll do once you're ok with the description.

**Comment by jforissier** [comment]
@etienne-lms crystal clear! Thanks for the details. For all commits:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
Tag applied.
